buildSrc/src/main/java/com/google/firebase/gradle/ReleaseGenerator.kt

```diff
@@ -0,0 +1,118 @@
+package com.google.firebase.gradle
+
+import java.io.File
+import java.io.IOException
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.api.ListBranchCommand
+import org.eclipse.jgit.api.errors.GitAPIException
+import org.eclipse.jgit.lib.Constants
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.Ref
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+
+open class ReleaseGenerator : DefaultTask() {
+    @TaskAction
+    @Throws(Exception::class)
+    fun generateReleaseConfig() {
+        val currentRelease = project.property("currentRelease").toString()
+        val pastRelease = project.property("pastRelease").toString()
+        val rootDir = project.rootDir
+        val availableModules = parseSubProjects(rootDir)
+
+        val repo = initRepo(rootDir)
+        val headRef = repo.repository.resolve(Constants.HEAD)
+        val branchRef = getObjectRefForBranchName(repo, pastRelease)
+
+        val changedDirs = getChangedLibraries(repo, branchRef, headRef, FIREBASE_LIBRARIES)
+        val changedModules = findChangedModules(changedDirs, availableModules)
+        writeReleaseConfig(rootDir, changedModules, currentRelease)
+    }
+
+    private fun findChangedModules(changedDirs: List<String>, availableModules: Set<String>): List<String> = changedDirs.flatMap { directory ->
+        val moduleName = directory.replace("/", ":")
+        val ktxModuleName = "$moduleName:ktx"
+        listOf(moduleName, ktxModuleName).filter { availableModules.contains(it) }
+    }
+
+    private fun parseSubProjects(rootDir: File) = File(rootDir, "subprojects.cfg").readLines().filter { !it.startsWith("#") }.toSet()
+
+    @Throws(IOException::class)
+    private fun initRepo(repoPath: File): Git {
+        if (!repoPath.exists()) {
+            throw RuntimeException("Repo path doesn't exist")
```

Comment:
FWIW, `Git.open` would throw the `IOException` if the file couldn't be found. I'm not sure what the error message would be though, so this may be preferable.

Reply:
Re-thinking this, this error is highly *highly* unlikely, given its pulled from project.rootDir, this line was needed when I was working on functionality

I'll remove this check in favor of the built in IOException